### DISCUSSION TITLE
How to send file data from frontend to Rust for FFMPEG processing

### DISCUSSION THREAD

**Discussion by iambenkay** [tauri-apps/tauri#7503] [Q&A] [answered]
I want to use an `input[type=file]` to load a video and then process this video with ffmpeg in the Rust backend. How can I convey this data in a way that does not impact the performance of my app?
The assumption is that these videos will be really large so I am skeptical about using FileReader to read the entire file into memory and pass it to an invoke call.
Is there a way I can pass a pointer to it over to Rust land and operate on this file pointer?

**Comment by FabianLars** [comment, marked as answer]
This is why tauri's own file picker dialog exists: https://tauri.app/v1/api/js/dialog. It will give you the file path instead of the file data itself. If you also want users to be able to drag and drop files then there's also tauri's own file drop event https://tauri.app/v1/api/js/window#onfiledropevent but it's a bit tricky to make this behave similar to the browser's native file drop events (it triggers on the whole window and conflicts with the native events)

**Reply by Lizitao610** [reply]
Another scenario is copying and pasting files, tauri doesn't seem to have a way to listen to paste files, what I do is to listen to the paste event on the front-end, and then notify rust, and get the contents of the pasteboard using rust.
